src/app/follow-button/follow.service.ts

```diff
@@ -0,0 +1,141 @@
+import { ChangeDetectorRef, Injectable} from "@angular/core";
```

Comment:
we should probably move this to live near the rest of the services instead of burying it in the follow-button directory